Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ILLD7YIGPositionCalibration.py

```diff
@@ -0,0 +1,517 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+
+from mantid.simpleapi import *
+from mantid.kernel import Direction, FloatBoundedValidator
+from mantid.api import FileAction, FileProperty, MatrixWorkspaceProperty, MultipleFileProperty, \
+    NumericAxis, Progress, PropertyMode, PythonAlgorithm, WorkspaceProperty
+from datetime import date
+import math
+import numpy as np
+import os
+from xml.dom import minidom
+import xml.etree.ElementTree as ET
+
+
+class ILLD7YIGPositionCalibration(PythonAlgorithm):
+
+    # helper conversions
+    _RAD_2_DEG = 180.0 / np.pi
+    _DEG_2_RAD = 1.0 / _RAD_2_DEG
+    # number of detectors
+    _D7NumberBanks = 3
+    _D7NumberPixels = 132
+    _D7NumberPixelsBank = 44
+    # an approximate universal peak width:
+    _peakWidth = 2.0 # in degrees
+    _beamMask1 = -35.0
+    _beamMask2 = 15.0
+    _minDistance = None
+    _scanStepSize = 0.25
+
+    def category(self):
+        return 'ILL\\Diffraction'
+
+    def summary(self):
+        return 'Performs D7 position calibration using YIG scan and returns D7 IPF readable by the NeXus data loader.'
+
+    def seeAlso(self):
+        return ['LoadILLPolarizedDiffraction']
+
+    def name(self):
+        return 'ILLD7YIGPositionCalibration'
+
+    def validateInputs(self):
+        issues = dict()
+        if (self.getProperty('Filenames').isDefault
+                and self.getProperty('ScanWorkspace').isDefault):
+            issues['Filenames'] = 'Either a list of filenames containing YIG scan \
+                or the workspace with the loaded scan is required for calibration.'
+            issues['ScanWorkspace'] = issues['Filenames']
+
+        return issues
+
+    def PyInit(self):
+        self.declareProperty(MultipleFileProperty("Filenames",
+                                                  action=FileAction.OptionalLoad),
+                             doc="The file names (including full or relative path) with a single YIG scan.")
```

Comment:
There can be more scans, for different wavelengths. This could be changed to 'The file names from a single YIG scan', or as you suggest 'The file names with a full YIG scan'.